react/react-data-fetching/react-data-fetching_iss-tracker/components/ISSTracker/index.js

```diff
@@ -1,45 +1,64 @@
 import { useEffect, useState } from "react";
 import Controls from "../Controls/index";
 import Map from "../Map/index";
+import useSWR from "swr";
 
 const URL = "https://api.wheretheiss.at/v1/satellites/25544";
 
+// We define a fetcher function that checks if the response is okay.
+// If the response is not okay, it throws an error with a message.
+// We use this fetcher function in the useSWR hook.
+const fetcher = async (url) => {
+  const response = await fetch(url);
+
+  if (!response.ok) {
+    const error = new Error(`Failed to fetch data: ${response.statusText}`);
+    error.info = await response.json();
+    error.status = response.status;
+    throw error;
+  }
+
+  const data = await response.json();
+  return data;
+};
+
+// We import useSWR from the SWR library.
+// We use the useSWR hook to fetch data from the specified URL ("https://api.wheretheiss.at/v1/satellites/25544").
+// useSWR provides us with data, error, isLoading, and isValidating as part of its response.
+// We use these states to handle loading and error states.
+// In the return statement, we directly use data.longitude and data.latitude instead of relying on a separate state.
+
 export default function ISSTracker() {
-  const [coords, setCoords] = useState({
-    longitude: 0,
-    latitude: 0,
-  });
-
-  async function getISSCoords() {
-    try {
-      const response = await fetch(URL);
-      if (response.ok) {
-        const data = await response.json();
-        setCoords({ longitude: data.longitude, latitude: data.latitude });
-      }
-    } catch (error) {
-      console.error(error);
+  // mutate function passed to the destructuring assignment from useSWR
+  const { data, error, isLoading, isValidating, mutate } = useSWR(
+    URL,
+    fetcher,
+    {
+      refreshInterval: 5000, // Set refresh interval to 5 seconds (in milliseconds)
     }
-  }
+  );
+
+  if (error) return <div>failed to load</div>;
```

Comment:
i wonder where those divs show up

Reply:
Only if the api cannot be fetched... since there are no errors (I assume), they do no show.